swift_scality_backend/diskfile.py

```diff
@@ -90,15 +84,20 @@ def put(self, metadata):
                          object
         """
         self._conn.send('0\r\n\r\n')
-        with contextlib.closing(self._conn):
+        try:
             resp = self._conn.getresponse()
+            msg = resp.read()
             if resp.status != 200:
-                msg = resp.read()
                 raise SproxydHTTPException("putting: %s / %s" % (
                     str(resp.status), str(msg)))
+        except Exception:
+            self._conn.close()
+            self._conn = None
```

Comment:
A somewhat better pattern might be:

```python
conn = self._conn
self._conn = None
try:
    conn.close()
except:
    logger.exception('Failure while closing connection')
raise
```
This assures `self._conn` will be set to `None`, even if the `close` operation fails for some reason.